core/src/test/java/org/apache/iceberg/avro/TestAvroDeleteWriters.java

```diff
@@ -0,0 +1,158 @@
+
+    ImmutableList.Builder<Record> builder = ImmutableList.builder();
+    builder.add(record.copy(ImmutableMap.of("id", 1L, "data", "a")));
+    builder.add(record.copy(ImmutableMap.of("id", 2L, "data", "b")));
+    builder.add(record.copy(ImmutableMap.of("id", 3L, "data", "c")));
+    builder.add(record.copy(ImmutableMap.of("id", 4L, "data", "d")));
+    builder.add(record.copy(ImmutableMap.of("id", 5L, "data", "e")));
+
+    this.records = builder.build();
+  }
+
+  @Test
+  public void testEqualityDeleteWriter() throws IOException {
+    File deleteFile = temp.newFile();
+
+    OutputFile out = Files.localOutput(deleteFile);
+    EqualityDeleteWriter<Record> deleteWriter = Avro.writeDeletes(out)
+        .createWriterFunc(DataWriter::create)
+        .overwrite()
+        .rowSchema(SCHEMA)
+        .withSpec(PartitionSpec.unpartitioned())
+        .equalityFieldIds(1)
+        .buildEqualityWriter();
+
+    try (EqualityDeleteWriter<Record> writer = deleteWriter) {
+      writer.deleteAll(records);
+    }
+
+    DeleteFile metadata = deleteWriter.toDeleteFile();
+    Assert.assertEquals("Format should be Avro", FileFormat.AVRO, metadata.format());
+    Assert.assertEquals("Should be equality deletes", FileContent.EQUALITY_DELETES, metadata.content());
+    Assert.assertEquals("Record count should be correct", records.size(), metadata.recordCount());
+    Assert.assertEquals("Partition should be empty", 0, metadata.partition().size());
+    Assert.assertNull("Key metadata should be null", metadata.keyMetadata());
+
+    List<Record> deletedRecords;
+    try (AvroIterable<Record> reader = Avro.read(out.toInputFile())
+        .project(SCHEMA)
+        .createReaderFunc(DataReader::create)
+        .build()) {
+      deletedRecords = Lists.newArrayList(reader);
+    }
+
+    Assert.assertEquals("Deleted records should match expected", records, deletedRecords);
+  }
+
+  @Test
+  public void testPositionDeleteWriter() throws IOException {
+    File deleteFile = temp.newFile();
+
+    Schema deleteSchema = new Schema(
+        MetadataColumns.DELETE_FILE_PATH,
+        MetadataColumns.DELETE_FILE_POS,
+        NestedField.optional(MetadataColumns.DELETE_FILE_ROW_FIELD_ID, "row", SCHEMA.asStruct()));
+
+    String deletePath = "s3://bucket/path/file.parquet";
+    GenericRecord posDelete = GenericRecord.create(deleteSchema);
+    List<Record> expectedDeleteRecords = Lists.newArrayList();
+
+    OutputFile out = Files.localOutput(deleteFile);
+    PositionDeleteWriter<Record> deleteWriter = Avro.writeDeletes(out)
+        .createWriterFunc(DataWriter::create)
+        .overwrite()
+        .rowSchema(SCHEMA)
+        .withSpec(PartitionSpec.unpartitioned())
+        .buildPositionWriter();
+
+    try (PositionDeleteWriter<Record> writer = deleteWriter) {
+      for (int i = 0; i < records.size(); i += 1) {
+        int pos = i * 3 + 2;
+        writer.delete(deletePath, pos, records.get(i));
```

Comment:
How about add an unit test to address the case that only write file/pos without `records.get(i)` for a delete marker ?   That's the positional row-delete case where we may don't want to reconstruct the streaming records.

Reply:
I agree, we should have a case where rows aren't passed.

Reply:
I added a test for this and I also updated the builder. Now, if you don't supply a row schema, the writer function for the row's object model is no longer required. And delete files will be written without the `row` column.